File: src/OpenFleet.Domain/Entities/Asset.cs
using OpenFleet.Domain.Enums;

namespace OpenFleet.Domain.Entities;

public class Asset : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string AssetTag { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty;
    public AssetCondition Condition { get; set; } = AssetCondition.Good;
    public AssetStatus Status { get; set; } = AssetStatus.Available;
    public DateTime PurchaseDate { get; set; }
    public Guid? DepartmentId { get; set; }
    public Guid? VehicleId { get; set; }

    public Department? Department { get; set; }
    public Vehicle? Vehicle { get; set; }
}

File: src/OpenFleet.Web/src/types/asset.ts
export type AssetStatus = 'Available' | 'InUse' | 'UnderMaintenance' | 'Decommissioned'

export type AssetCondition = 'New' | 'Good' | 'Fair' | 'Poor' | 'Damaged'

export interface AssetResponse {
  id: string
  assetTag: string
  name: string
  type: string
  condition: AssetCondition
  status: AssetStatus
  departmentId: string | null
  departmentName: string | null
  vehicleId: string | null
  vehicleDescription: string | null
  purchaseDate: string
  createdAt: string
  updatedAt: string
}

export interface CreateAssetRequest {
  assetTag: string
  name: string
  type: string
  condition: AssetCondition
  status: AssetStatus
  departmentId: string
  vehicleId?: string | null
}

export interface UpdateAssetRequest {
  assetTag?: string
  name?: string
  type?: string
  condition?: AssetCondition
  status?: AssetStatus
  departmentId?: string
  vehicleId?: string | null
}

export interface AssetFilterRequest {
  status?: AssetStatus
  type?: string
  condition?: AssetCondition
  departmentId?: string
  vehicleId?: string
}
